recipes/newrelic/apm/java/linux.yml

```diff
@@ -301,22 +303,54 @@ install:
-                    else
                       echo $(build_stdout_json "Attaching agent to Tomcat process in '$dockerAppName'...complete!")
+                      # add deployment marker
+                      appId=$(curl -sX GET "DEFAULT_API_URL/v2/applications.json" -H "Api-Key:${api_key}" -d "filter[name]=${dockerAppName}" | sed 's/.*"id":"\{0,1\}\([^,"]*\)"\{0,1\}.*/\1/')
+                      >/dev/null curl -sX POST "DEFAULT_API_URL/v2/applications/$appId/deployments.json" \
+                      -H "Api-Key:${api_key}" \
+                      -H "Content-Type: application/json" \
+                      --data-binary @- << EOF
+                      {
+                        "deployment": {
+                        "revision": "JAVA-AGENT-ATTACHMENT",
+                        "changelog": "/etc/newrelic-infra/integrations.d/java-dynamic-attach.yml attached java agent to PID ${JAVA_PID} for APM Service '${dockerAppName}'",
+                        "user": "/etc/newrelic-java/dynamic-attach.sh",
+                        "timestamp": "${deploymentTimestamp}"
+                        }
+                      }
+          EOF
+                    else
+                      echo $(build_stdout_json "Attaching agent to Tomcat process in '$dockerAppName'...failed!")
                     fi
                   else
                     echo $(build_stdout_json "Skipping instrumentation of process id $JAVA_PID: could not determine owning container")
                   fi
                 else
                   echo $(build_stdout_json "Attaching agent to Tomcat process with PID $JAVA_PID...")
-                  output=$(sudo nri-lsi-java -apm $JAVA_PID -appName $appName -license $license -region $region 2> /dev/null | wc -l)
+                  grepPattern='\"level\":\"INFO\",\"label\":\"Url\"'
+                  output=$(sudo nri-lsi-java -apm $JAVA_PID -appName $appName -license $license -region $region | grep ${grepPattern} | grep -v grep  | wc -l)
                   if [[ $output -gt 0 ]]; then
-                    echo $(build_stdout_json "Attaching agent to Tomcat process with PID $JAVA_PID...failed!")
-                  else
                     echo $(build_stdout_json "Attaching agent to Tomcat process with PID $JAVA_PID..complete!")
+                    # add deployment marker
+                    appId=$(curl -sX GET "DEFAULT_API_URL/v2/applications.json" -H "Api-Key:${api_key}" -d "filter[name]=${appName}" | sed 's/.*"id":"\{0,1\}\([^,"]*\)"\{0,1\}.*/\1/')
+                    >/dev/null curl -sX POST "DEFAULT_API_URL/v2/applications/$appId/deployments.json" \
+                    -H "Api-Key:${api_key}" \
+                    -H "Content-Type: application/json" \
+                    --data-binary @- << EOF
+                    {
+                      "deployment": {
+                      "revision": "JAVA-AGENT-ATTACHMENT",
```

Comment:
@mbruzina 
maybe just JAVA-AGENT-ATTACH or JAVA-DYNAMIC-ATTACH to keep consistency.